### DISCUSSION TITLE
Connection refuesed with Gmail SMTP

### DISCUSSION THREAD

**Discussion by chucky5150** [authelia/authelia#7445] [Q&A] [answered]
I can get the notifier working with a standard text file. 
However, I can not seem to get it working via Google's SMTP server. Went back and copied the code from the "Using Gmail" page. 

> failed to dial connection: dial tcp 142.250.113.108:587: connect: connection refused

I've tried searching the googles and other config files that I've come across to see how it was done. I even tried turning UFW off to see if that made any difference. I am able to ping 142.250.113.108. 

```
notifier:
  smtp:
    username: '[myaccount]@gmail.com'
    # Password can also be set using a secret: https://www.authelia.com/configuration/methods/secrets/
    password: '[app-password]'
    sender: '[myaccount]@gmail.com'
    host: 'smtp.gmail.com'
    port: 587
```

Log from Authelia:
```
time="2024-06-20T08:49:08-05:00" level=info msg="Authelia v4.38.8 is starting"

time="2024-06-20T08:49:08-05:00" level=info msg="Log severity set to debug"

time="2024-06-20T08:49:11-05:00" level=info msg="Storage schema is being checked for updates"

time="2024-06-20T08:49:11-05:00" level=info msg="Storage schema is already up to date"

time="2024-06-20T08:49:12-05:00" level=error msg="Error occurred running a startup check" error="failed to dial connection: dial tcp 142.250.113.108:587: connect: connection refused" provider=notification stack="github.com/authelia/authelia/v4/internal/commands/root.go:128 doStartupChecks\ngithub.com/authelia/authelia/v4/internal/commands/root.go:88  (*CmdCtx).RootRunE\ngithub.com/spf13/cobra@v1.8.0/command.go:983                  (*Command).execute\ngithub.com/spf13/cobra@v1.8.0/command.go:1115                 (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.8.0/command.go:1039                 (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10       main\nruntime/internal/atomic/types.go:194                          (*Uint32).Load\nruntime/asm_amd64.s:1695                                      goexit"

time="2024-06-20T08:49:12-05:00" level=fatal msg="One or more providers had fatal failures performing startup checks, for more detail check the error level logs" providers="[notification]" stack="github.com/authelia/authelia/v4/internal/commands/root.go:150 doStartupChecks\ngithub.com/authelia/authelia/v4/internal/commands/root.go:88  (*CmdCtx).RootRunE\ngithub.com/spf13/cobra@v1.8.0/command.go:983                  (*Command).execute\ngithub.com/spf13/cobra@v1.8.0/command.go:1115                 (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.8.0/command.go:1039                 (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10       main\nruntime/internal/atomic/types.go:194                          (*Uint32).Load\nruntime/asm_amd64.s:1695                                      goexit"
```

I deals on what else I can try?

**Comment by glime-ai** [comment, marked as answer]
Please note this response was created using generative AI leveraging previously available information from the repository. This information may be helpful however it is our strong recommendation that you carefully consider the information before blindly applying it.

---

It looks like you're encountering a network issue when trying to connect to Google's SMTP server. Here are a few steps you can take to troubleshoot and resolve this issue:

1. **Check Network Connectivity**: Ensure that your server has internet access and can reach Google's SMTP server. You can use tools like `telnet` or `nc` to test connectivity to `smtp.gmail.com` on port `587`.

    ```sh
    telnet smtp.gmail.com 587
    ```

    If the connection is refused, there might be a network issue or firewall blocking the connection.

2. **Firewall and Security Groups**: Ensure that there are no firewall rules or security groups blocking outbound connections on port `587`. Since you mentioned turning off UFW, double-check any other network security settings that might be in place.

3. **DNS Resolution**: Verify that your server can resolve `smtp.gmail.com` to the correct IP address. You can use `nslookup` or `dig` to check this.

    ```sh
    nslookup smtp.gmail.com
    ```

4. **Gmail SMTP Configuration**: Ensure that your Gmail account is configured to allow SMTP connections. You might need to enable "Less secure app access" or generate an app-specific password if you have 2FA enabled.

5. **Authelia Configuration**: Double-check your Authelia configuration to ensure all settings are correct. Here is an example configuration for using Gmail's SMTP server:

    ```yaml
    notifier:
      smtp:
        username: 'myaccount@gmail.com'
        password: 'yourapppassword'
        sender: 'myaccount@gmail.com'
        host: 'smtp.gmail.com'
        port: 587
    ```

6. **Logs and Debugging**: Review the Authelia logs for any additional error messages or clues. You can enable more verbose logging if needed.

If you continue to experience issues, you might want to provide more detailed logs or consider reaching out to your network administrator to ensure there are no network-level blocks.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=6fsHkcyufjQLaB&utm_medium=github).

**Reply by chucky5150** [reply]
ok, seems like it's a VPS issue. Ran the `telnet smtp.gmail.com 587` on the remote server and connection was refused. Ran the same code from my desktop and it seems to work.
